view/src/main/java/cmm/apps/esmorga/view/eventdetails/EventDetailsViewModel.kt

```diff
@@ -56,8 +50,43 @@ class EventDetailsViewModel(
     }
 
     fun onPrimaryButtonClicked() {
-        if (!isAuthenticated) {
+        if (isAuthenticated) {
+            if (userJoined) {
+                // TODO Leave event
+            } else {
+                joinEvent()
+            }
+        } else {
             _effect.tryEmit(EventDetailsEffect.NavigateToLoginScreen)
         }
     }
+
+    private fun getEventDetails() {
+        viewModelScope.launch {
+            val user = getSavedUserUseCase()
+            val result = getEventDetailsUseCase(eventId)
+            isAuthenticated = user.data != null
+            result.onSuccess {
+                userJoined = it.userJoined
+                _uiState.value = it.toEventUiDetails(isAuthenticated, userJoined)
+            }
+        }
+    }
+
+    private fun joinEvent() {
+        viewModelScope.launch {
+            _uiState.value = _uiState.value.copy(primaryButtonLoading = true)
+            val result = joinEventUseCase(eventId)
+            result.onSuccess {
+                getEventDetails()
```

Comment:
No need to get the event again from local. You know already that the user has joined, because the result is success.